src/main/java/com/fasterxml/jackson/databind/introspect/JacksonAnnotationIntrospector.java

```diff
@@ -1510,7 +1510,13 @@ protected TypeResolverBuilder<?> _findTypeResolver(MapperConfig<?> config,
     {
         // since 2.16 : backporting {@link JsonTypeInfo.Value} from 3.0
         JsonTypeInfo.Value typeInfo = findPolymorphicTypeInfo(config, ann);
-
+        // [databind#3943] Add config-override system for JsonTypeInfo.Value
+        JsonTypeInfo.Value overrideTypeInfo = config.getDefaultPolymorphicTypeHandling(
```

Comment:
Ideally this should not be done by `AnnotationInstropector`s, but something that calls them; is there any possibility of doing that? The idea is that (in theory at least) AI only accesses pieces of annotation and similar. It might be worth looking at how other settings from `ConfigOverride` are used, called from `SerializerProvider` and/or `DeserializationContext` (or maybe `SerializationConfig`/`DeserializationConfig`).

My main concern is with overrides only being applied for Jackson's `@JsonTypeInfo`, and not overriding information from JAXB equivalents.

But I am also not sure how feasible handling outside this location is.